core/test/EventWorkspace_test.cpp

```diff
@@ -119,32 +119,40 @@ TEST(EventWorkspace, basics) {
 }
 
 TEST(EventWorkspace, plus) {
+  // Note that unlike the tests above this is now using sparse dimensions.
+  // Addition for nested Dataset as event list is not supported anymore.
   Dataset d;
-
-  Dataset e;
-  e.insert(Data::Tof, "", {Dim::Event, 10});
-  e.insert(Data::PulseTime, "", {Dim::Event, 10});
-  auto e2 = concatenate(e, e, Dim::Event);
-
-  d.insert(Data::Events, "", {Dim::Spectrum, 2}, {e, e2});
-
-  EXPECT_THROW_MSG(d - d, std::runtime_error,
-                   "Subtraction of events lists not implemented.");
-  EXPECT_THROW_MSG(d * d, std::runtime_error,
-                   "Multiplication of events lists not implemented.");
-
-  // Special handling: Adding datasets *concatenates* the event lists.
-  auto sum = d + d;
-
-  auto eventLists = sum.get(Data::Events);
-  EXPECT_EQ(eventLists.size(), 2);
-  EXPECT_EQ(eventLists[0].get(Data::Tof).size(), 2 * 10);
-  EXPECT_EQ(eventLists[1].get(Data::Tof).size(), 2 * 20);
-
-  sum += d;
-
-  eventLists = sum.get(Data::Events);
-  EXPECT_EQ(eventLists.size(), 2);
-  EXPECT_EQ(eventLists[0].get(Data::Tof).size(), 3 * 10);
-  EXPECT_EQ(eventLists[1].get(Data::Tof).size(), 3 * 20);
+  d.insert(
+      makeSparseVariable<double>(Coord::Tof, {Dim::Spectrum, 2}, Dim::Tof));
+  auto tofs = d(Coord::Tof).sparseSpan<double>();
+  tofs[0].resize(10);
+  tofs[1].resize(20);
+  d.insert(
+      makeSparseVariable<double>(Data::Value, {Dim::Spectrum, 2}, Dim::Tof));
+  auto weights = d(Data::Value).sparseSpan<double>();
```

Comment:
As separate variables there is no conceptional connection (as far as dataset is concerned) between sparse variables. weights and tofs in this case? i.e dataset doesn't attempt to do anything like stopping you have more weights than tofs?

Reply:
Indeed. We have to support different sparse extents for cases like having sample and vanadium data. However, #234 proposes a fundamental change, in effect grouping all connected sparse variables into a single "variable", which could then enforce aligned dimensions. I am saying could, since it may still turn out to be impractical to enforce this.